language-support/ts/daml-react/credentials.ts

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2020 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { decode } from 'jwt-simple';
+
+export type Credentials = {
+  party: string;
+  token: string;
+  ledgerId: string;
+}
+
+/**
+ * Check that the party in the token matches the party of the credentials and
+ * that the ledger ID in the token matches the given ledger id.
+ */
+export const preCheckCredentials = ({party, token, ledgerId}: Credentials): string | null => {
+  const decoded = decode(token, '', true);
+  if (!decoded.ledgerId || decoded.ledgerId !== ledgerId) {
+    return 'The password is not valid for DAVL.';
```

Comment:
This also mentions DAVL. If there is more stuff that references DAVL I’m happy for this to be addressed in a separate PR. probably good to just grep through the codebase before we publish the library.

Reply:
thanks, fixed.